File: api/Core/Model/ConfigSetting.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Xcianify.Core.Model
{
    public class ConfigSetting : BaseModel
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string SettingKey { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(150)]
        public string SettingName { get; set; } = string.Empty;
        
        public string? Description { get; set; }
        
        public string? StringValue { get; set; }
        
        public long? IntegerValue { get; set; }
        
        public bool? BooleanValue { get; set; }
        
        public decimal? DecimalValue { get; set; }
        
        public string? DefaultValue { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        public int? CreatedBy { get; set; }
        
        public int? UpdatedBy { get; set; }
    }
} 

File: pharma-erp-3/types/config-setting.ts
export interface ConfigSetting {
  id: number;
  settingKey: string;
  settingName: string;
  description?: string;
  stringValue?: string;
  integerValue?: number;
  booleanValue?: boolean;
  decimalValue?: number;
  defaultValue?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  createdBy?: number;
  updatedBy?: number;
}

export type CreateConfigSettingData = Omit<ConfigSetting, 'id' | 'createdAt' | 'updatedAt' | 'createdBy' | 'updatedBy'>;

export type UpdateConfigSettingData = Partial<CreateConfigSettingData> & { id: number };

export interface ConfigSettingFilter {
  search?: string;
  page?: number;
  pageSize?: number;
  isActive?: boolean;
} 